### PR TITLE
upgrade to go 1.23.1

### PR DESCRIPTION
**Pull request by BrainGithub**
I suggest merge this to keep updated.

### CODE REVIEW

**Comment by BrainGithub** [comment]
@vcastellm Can you please help review, I can run localtest passed except for the AgentCommand for Election, I think that could be becouse of the macOS ENV issue, I suggest merge this to keep updated, because from any aspects the go v1.16 is too old.
`=== RUN   TestAgentCommand_runForElection
panic: test timed out after 10m0s
	running tests:
		TestAgentCommand_runForElection (10m0s)

goroutine 34 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/Cellar/go/1.23.0/libexec/src/testing/testing.go:2373 +0x385
created by time.goFunc
	/usr/local/Cellar/go/1.23.0/libexec/src/time/sleep.go:215 +0x2d

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0xc000c14820, {0x7177393?, 0xba8600000067b50?}, 0x87c4b70)
	/usr/local/Cellar/go/1.23.0/libexec/src/testing/testing.go:1751 +0x3ab
testing.runTests.func1(0xc000c14820)
	/usr/local/Cellar/go/1.23.0/libexec/src/testing/testing.go:2168 +0x37
testing.tRunner(0xc000c14820, 0xc000067c70)
	/usr/local/Cellar/go/1.23.0/libexec/src/testing/testing.go:1690 +0xf4
testing.runTests(0xc000c03cb0, {0x9b21c20, 0x32, 0x32}, {0x57c5110?, 0x57c4d7a?, 0x9b4fa80?})
	/usr/local/Cellar/go/1.23.0/libexec/src/testing/testing.go:2166 +0x43d
testing.(*M).Run(0xc000199680)
	/usr/local/Cellar/go/1.23.0/libexec/src/testing/testing.go:2034 +0x64a
main.main()
	_testmain.go:143 +0x9b

goroutine 20 [runnable]:
go.opencensus.io/stats/view.(*worker).start(0xc00050f000)
	/Volumes/newVolumes/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:292 +0x9f
created by go.opencensus.io/stats/view.init.0 in goroutine 1
	/Volumes/newVolumes/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:34 +0x8d

goroutine 7 [runnable]:
syscall.syscall(0xc001180000?, 0xc0004e4a40?, 0x5a31155?, 0x57c1bbd?)
	/usr/local/Cellar/go/1.23.0/libexec/src/runtime/sys_darwin.go:27 +0x58
syscall.bind(0x2775?, 0x0?, 0x0?)
	/usr/local/Cellar/go/1.23.0/libexec/src/syscall/zsyscall_darwin_amd64.go:74 +0x2e
syscall.Bind(0x4, {0x87d0720?, 0xc001180000?})
	/usr/local/Cellar/go/1.23.0/libexec/src/syscall/syscall_unix.go:292 +0x46
net.(*netFD).listenStream(0xc00110c480, {0x87efcb0, 0x9b74120}, {0x87f4f18, 0xc001106390}, 0x80, 0x0)
	/usr/local/Cellar/go/1.23.0/libexec/src/net/sock_posix.go:167 +0x225
net.socket({0x87efcb0, 0x9b74120}, {0x711d9d1, 0x4}, 0x2, 0x1, 0x5a2f70b?, 0x0, {0x87f4f18, 0xc001106390}, ...)
	/usr/local/Cellar/go/1.23.0/libexec/src/net/sock_posix.go:57 +0x1b4
net.internetSocket({0x87efcb0, 0x9b74120}, {0x711d9d1, 0x4}, {0x87f4f18, 0xc001106390}, {0x0, 0x0}, 0x1, 0x0, ...)
	/usr/local/Cellar/go/1.23.0/libexec/src/net/ipsock_posix.go:167 +0xf8
net.(*sysListener).listenTCPProto(0xc0004e4cf8, {0x87efcb0?, 0x9b74120?}, 0x0?, 0x0?)
	/usr/local/Cellar/go/1.23.0/libexec/src/net/tcpsock_posix.go:189 +0xa5
net.(*sysListener).listenTCP(...)
	/usr/local/Cellar/go/1.23.0/libexec/src/net/tcpsock_posix.go:179
net.ListenTCP({0x711d9d1, 0x4}, 0x10?)
	/usr/local/Cellar/go/1.23.0/libexec/src/net/tcpsock.go:442 +0x105
github.com/hashicorp/serf/testutil.TakeIP()
	/Volumes/newVolumes/go/pkg/mod/github.com/hashicorp/serf@v0.10.1/testutil/addr.go:57 +0xde
github.com/distribworks/dkron/v3/dkron.TestAgentCommand_runForElection(0xc000c149c0)
	/Volumes/newVolumes/go/src/github.com/dkron/dkron/agent_test.go:29 +0xc8
testing.tRunner(0xc000c149c0, 0x87c4b70)
	/usr/local/Cellar/go/1.23.0/libexec/src/testing/testing.go:1690 +0xf4
created by testing.(*T).Run in goroutine 1
	/usr/local/Cellar/go/1.23.0/libexec/src/testing/testing.go:1743 +0x390
FAIL	github.com/distribworks/dkron/v3/dkron	604.543s`

File: Makefile [hunk 1]
```diff
@@ -89,7 +89,8 @@ plugin/types/%.pb.go: proto/%.proto
 
 ui: dkron/ui-dist
 
-main: dkron/ui-dist plugin/types/dkron.pb.go plugin/types/executor.pb.go *.go */*.go */*/*.go */*/*/*.go
+#main: dkron/ui-dist plugin/types/dkron.pb.go plugin/types/executor.pb.go *.go */*.go */*/*.go */*/*/*.go
```
Comment (vcastellm) [inline Makefile:92]: Remove comment?

**Comment by vcastellm** [comment]
@BrainGithub tests are passing in the CI

**Review by BrainGithub — COMMENTED** [review]
Look good to me.

File: Makefile (hunk 1 above)
Comment (BrainGithub) [reply Makefile]: It's for local test. 
And sorry that I missed this message, a new pull request, please take a review.

**Review by vcastellm — APPROVED** [review]
LGTM thanks!

**Pull request merged**